### PR TITLE
Turborepo improvements

### PR DESCRIPTION
**Pull request by zomars**
## What does this PR do?

Splits common tasks into individual ones. This will allow us to take advantage of caching and remove redundant task runs.

refs #3136

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- [ ] `yarn dx`
- [ ] `yarn dev`
- [ ] `yarn test-e2e`
- [ ] `yarn build`
- [ ] `yarn start`
- [ ] @hariombalhara please check if `yarn turbo run embed-tests-update-snapshots:ci --scope=@calcom/embed-react --concurrency=1` works as expected

### CODE REVIEW

File: turbo.json [hunk 1]
```diff
@@ -156,8 +125,22 @@
       "dependsOn": ["@calcom/prisma#db-reset", "@calcom/web#test", "@calcom/web#build"]
     },
     "type-check": {
-      "dependsOn": ["@calcom/embed-react#build", "@calcom/embed-core#build", "@calcom/embed-snippet#build"],
+      "cache": false,
       "outputs": []
+    },
```
Comment (zomars) [inline turbo.json:130]: Type checking should be able to run without dependencies

File: turbo.json [hunk 2]
```diff
@@ -78,44 +60,27 @@
...
-      ],
+      "dependsOn": ["^build"],
       "outputs": ["dist/**", ".next/**"]
     },
-    "db-deploy": {},
-    "db-seed": {},
+    "db-deploy": {
+      "inputs": ["./migrations/**/*.sql", "./prisma/migrations/**/*.sql"],
+      "outputs": []
+    },
```
Comment (zomars) [inline turbo.json:82]: Run production migrations when there are new ones

File: turbo.json [hunk 3]
```diff
@@ -1,48 +1,47 @@
...
     "@calcom/prisma#db-reset": {
-      "cache": false
+      "cache": false,
+      "dependsOn": [
+        "@calcom/prisma#db-nuke",
+        "@calcom/prisma#db-up",
+        "@calcom/prisma#db-deploy",
+        "@calcom/prisma#db-seed"
+      ],
+      "outputs": []
```
Comment (zomars) [inline turbo.json:26]: Should be run with `--concurrency=1` since the order matters here.

File: turbo.json [hunk 4]
```diff
@@ -125,10 +90,13 @@
       "cache": false
     },
     "dev": {
+      "dependsOn": ["//#env-check:common", "//#env-check:app-store"],
+      "outputs": [],
       "cache": false
     },
```
Comment (zomars) [inline turbo.json:96]: We check the env variables before dev if the example files changed

File: turbo.json [hunk 5]
```diff
@@ -139,7 +107,8 @@
       "outputs": ["lint-results"]
     },
     "post-install": {
-      "cache": false
+      "outputs": ["../../node_modules/@prisma/client/**", "../../node_modules/@prisma/admin-client/**"],
+      "inputs": ["./schema.prisma", "./prisma/schema.prisma"]
     },
```
Comment (zomars) [inline turbo.json:112]: Regenerate prisma clients on post-install client if schemas change

**Review by agustif — APPROVED** [review]
Awesome results @zomars, `yarn dx` feels really fast now, `yarn start` is instant, `yarn build`, took 70secs which ain't bad.

e2e is failing for me though?
image

**Review by agustif — CHANGES_REQUESTED** [review]
e2e are failing for me. unapproving until I found out why. Seems CI has the same issue @zomars , is this OK?

**Review by jaredpalmer — APPROVED** [review]

**Pull request merged**